### PR TITLE
feat: Docker

### PR DESCRIPTION
**Pull request by FranciscoJBrito**
### Features
1 - To reduce the size of the Dockerfile I used an Alpine image of Node.
__THE MOST OF THE CONTRIBUTION MADE IN THE DOCKERFILE BELONGS TO @Geekboy1011__

2 - Added the `entrypoints.sh` file, to run the yarn commands.

3 - Additionally add the docker-compose file to include redis, because otherwise you would get an error when running the backend. For it to work correctly, the variable `REDIS_HOST=host.docker.internal:6379` must be added to the .env file

4- Added .dockerignore to reduce the size of the Dockerfile as much as possible

There are still areas for improvement, but this first version works.

---
### Reference
#179

### CODE REVIEW

**Comment by hudy9x** [comment]
I have some comments as follows
- backend and frontend should be ran separately, it means we need one service for backend and the other for frontend
- redis service is ok
- need a database service
- need environment sections in docker-compose file
for now, everything seem to be good

**Comment by FranciscoJBrito** [comment]
@hudy9x great, we take these points as future modifications or would they be required for this PR to be merged?

**Comment by hudy9x** [comment]
@FranciscoJBrito 
>we take these points as future mods or would they be necessary to merge this PR?

It should be resolved this PR

>In case they are needed for the merge, let me know what environment variables you want to add. I am currently passing the .env.local file as env_file argument in docker-compose.

For now, just use your version

>On the other hand, the database you mentioned adding as a service in docker-compose, do you mean adding mongodb

yes

**Comment by FranciscoJBrito** [comment]
@hudy9x for now, I prefer not to add mongo as a service in docker-compose, I did some tests and the app performs better if the connection is with Atlas.

**Comment by hudy9x** [comment]
@FranciscoJBrito 
I see `build` field in backend service, but the frontend service does.
Based on my limited Docker experience, there should be an additional "build" field and Docker file in the docker-compose
If you're not planned to implement the frontend service this time, please remove it. And leave Backend + Redis service.

P/S: I wonder that could we put all files that relate to Docker into a folder `/dockers` ?

**Comment by FranciscoJBrito** [comment]
@hudy9x the build field is for building the Dockerfile, if I added the build in both services what I would be doing is building the image twice, which is unnecessary and a waste of resources. So what I do is build the image once on the backend, and then on the frontend I use the image previously built in the backend service. In conclusion, rest assured that it is the same image in both services, we build it only once and use it twice.

Regarding adding the files inside a /docker folder, of course, it is a good idea to maintain organization, the only files I have to leave in the root are the docker-compose and .dockerignore, but the rest I will add in the /docker folder.

**Comment by hudy9x** [comment]
@FranciscoJBrito 
Ok, i got it. I'll merge this PR once you complete the updates.

File: docker/entrypoints.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# This script is used to start the application.
+
+echo "Starting application..."
+
+# Check if Prisma commands have already been executed
+if [ ! -f "/app/.prisma_initialized" ]; then
```
Comment (hudy9x) [inline docker/entrypoints.sh:8]: forgot this from the prev review, but you fixed it now. 👍

**Pull request merged**